Kernel/Process.cpp

```diff
@@ -76,9 +76,9 @@ UNMAP_AFTER_INIT void Process::initialize()
 
     next_pid.store(0, AK::MemoryOrder::memory_order_release);
 
-    hostname().with_exclusive([&](auto& name) {
-        name = "courage";
-    });
+    // Note: This is called before scheduling is initialized, and before APs are booted.
+    //    So we can "safely" bypass the lock here.
+    reinterpret_cast<String&>(hostname()) = "courage";
```

Comment:
dumb comment request, please align that "So" with "This", it's surprisingly irritating to me...

p.s. I don't like the fact that this is allowed, just casting to the held type is too easy of a bypass :(

Reply:
alternative for this would be to add a `hostname().with______out_the_lock_because_Im_in_early_boot_I_pinky_swear()` API or somethin

Reply:
I'm not asking for a method, I'm saying "reinterpret_cast the lock away" is too easy
Maybe `*ProtectedValue<T>` could have a `T` member instead of inheriting from `T` (as its first member)?
Then you could `bit_cast<String&>(&hostname())` to get it, knowing that what you're doing is horrible and you should go chill in the shame box for a while.

Reply:
Problem is, all we really want to do initialize this thing to a default value before we start any threads.  If we allowed global constructors... that would be easy :)

Reply:
with that, you'd be perfectly able to `shame_box.contents = move(self);`
either way though, that's not really related to this PR.

Reply:
wait wait wait, `WhateverProtected<T>` _doesn't_ inherit from `T`, why is this `reinterpret_cast` even allowed?

Reply:
K i moved the comment. Alignment > 4 spaces it seems :)